src/components/modals/QrCodeModal.tsx

```diff
@@ -127,30 +130,38 @@ class QrCodeModalComponent extends React.Component<Props, State> {
     this.setState({ pendingLogin: out })
   }
 
+  handleCancel = (): void => {
+    const { bridge } = this.props
+    bridge.resolve(undefined)
+  }
+
   render() {
     const { bridge, theme } = this.props
     const { pendingLogin, username } = this.state
     const styles = getStyles(theme)
-
+    const isLoadingQrCode = username != null || pendingLogin == null
+    const qrData =
+      pendingLogin == null ? null : 'edge://edge/' + pendingLogin.id
     return (
-      <ThemedModal bridge={bridge} onCancel={() => bridge.resolve()}>
-        <TitleText>{s.strings.qr_modal_title}</TitleText>
-        <MessageText>
-          {username != null
-            ? sprintf(s.strings.qr_modal_started, username)
-            : s.strings.qr_modal_message}
-        </MessageText>
-        <View style={styles.qrContainer}>
-          {username != null || pendingLogin == null ? (
-            <ActivityIndicator color={theme.primaryText} />
-          ) : (
-            <QrCode
-              data={'edge://edge/' + pendingLogin.id}
-              size={theme.rem(14)}
-            />
-          )}
-        </View>
-        <ModalCloseArrow onPress={() => bridge.resolve()} />
+      <ThemedModal bridge={bridge} onCancel={this.handleCancel}>
+        <ModalScrollArea onCancel={this.handleCancel}>
```

Comment:
I don't think the QR modal needs to scroll? The QR code will just shrink to fit, so this shouldn't be necessary.

Reply:
It could be helpful for the user to be able to vertically translate it in some circumstances such as if are using two apps simultaneously on Android, or if the users device screen has been cracked from the bottom left/right corner, and repositioning the QR code slightly higher allows for a successful scan.

Reply:
Ok, it's weird but we can go with it.